core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -4559,8 +4564,8 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
       sendRequestsLatch.await()
       verify(blockStoreClient, times(2))
         .finalizeShuffleMerge(any(), any(), any(), any(), any())
-      assert(sentHosts.nonEmpty)
-      assert(sentHosts.head === "hostB" && sentHosts.length == 1)
+      assert(1 == sentHosts.size())
+      assert(sentHosts.asScala.toSeq === Seq("hostB"))
```

Comment:
In the test failures seen earlier, `verify(blockStoreClient, times(2))` was satisfied. The reason why `sentHosts` was empty is due to the race in update between `sendRequestsLatch` and `sentHosts`, and also because `sentHosts` was not thread safe (so `add` to it need to show in this thread)